modules/javafx.graphics/src/test/java/test/javafx/scene/Node_lookup_Test.java

```diff
@@ -96,4 +109,89 @@ public class Node_lookup_Test {
         assertTrue(nodes.contains(d));
         assertTrue(nodes.contains(d2));
     }
+
+    @Test
+    public void lookupPseudoTest() {
+        Set<Node> nodes = root.lookupAll(".h:testPseudo2");
+        assertEquals(1, nodes.size());
+        assertTrue(nodes.contains(hg));
+
+        Node found = root.lookup(".h:testPseudo2");
+        assertSame(hg, found);
+
+        found = root.lookup(":testPseudo2");
+        assertSame(hg, found);
+
+        found = root.lookup(".h:testPseudo1:testPseudo2");
+        assertSame(hg, found);
+
+        nodes = root.lookupAll(".g:testPseudo1");
+        assertEquals(2, nodes.size());
+        assertTrue(nodes.contains(g));
+        assertTrue(nodes.contains(hg));
+
+        nodes = root.lookupAll(":testPseudo1");
+        assertEquals(2, nodes.size());
+        assertTrue(nodes.contains(g));
+        assertTrue(nodes.contains(hg));
+
+        nodes = root.lookupAll(".f > .h:testPseudo1");
+        assertEquals(1, nodes.size());
+        assertTrue(nodes.contains(hg));
+
+        nodes = root.lookupAll(".f:testPseudo > .h:testPseudo1");
+        assertEquals(1, nodes.size());
+        assertTrue(nodes.contains(hg));
+
+        nodes = root.lookupAll(".f:randomPseudo");
+        assertEquals(0, nodes.size());
+    }
+
+    /**
+     * Verifies that the lookup ignores pseudo classes when selector contains no explicit pseudo class.
+     */
+    @Test
+    public void lookupPseudoTest2() {
```

Comment:
the reason I've used different nodes is because the situation you have with existing nodes might give false positive.

I wanted to make sure that selector "a." will collect nodes with ".a" and ".a:pseudo", but in the standard set of nodes with g and hg nodes - but they have the same :testPseudo1, so it's not exactly equivalent.

The other option is to add new nodes, one with ".x" and another with ".x:pseudo"
If you do that, it might be easier to change the javadoc format to do something like this:

```
 └─ Pane
     ├─ VFlow (Pane) .flow
     │   ├─ ClippedPane (Pane) .content
```
(but it's just a very minor suggestion)

Reply:
Thanks for the clarification. Now I got the point what you are saying about.
Added a new test case to cover this scenario. Used x, y.. naming to make a clear separation from previous scenarios.

However do you still prefer to keep the lookupPseudoTest2() scenarios? This is almost similar to quickTest() and lookupAllTest() except that all the nodes used in this method have pseudo classes set.

Reply:
we could keep it - the more the merrier.

ideally, we could use a more thorough approach: let's say, use two or three layers, and create as many nodes as there are combinations, to test every combination of the selector and search result.

but the main result we got is that there should be no regression w.r.t. the old implementation (I think).
two more people should definitely take a look at this PR still.

Reply:
Ok. Thank you :)